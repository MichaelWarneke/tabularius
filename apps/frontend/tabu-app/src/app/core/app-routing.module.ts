import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
