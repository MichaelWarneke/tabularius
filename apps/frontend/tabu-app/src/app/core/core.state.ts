import {
  ActionReducerMap,
  MetaReducer,
  createFeatureSelector
} from '@ngrx/store';
import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { RouterStateUrl } from '@tabularius/shared/services';
import { storeFreeze } from 'ngrx-store-freeze';

import { initStateFromLocalStorageReducer } from '@tabularius/shared/services';
import { debugReducer } from '@tabularius/shared/services';

import { environment } from '../../environments/environment';

export interface AppState {
  router: RouterReducerState<RouterStateUrl>;
}

export const reducers: ActionReducerMap<AppState> = {
  router: routerReducer
};

export const metaReducers: MetaReducer<AppState>[] = [
  initStateFromLocalStorageReducer
];
if (!environment.production) {
  metaReducers.unshift(storeFreeze);
  metaReducers.unshift(debugReducer);
}

export const selectRouterState = createFeatureSelector<
  AppState,
  RouterReducerState<RouterStateUrl>
>('router');
