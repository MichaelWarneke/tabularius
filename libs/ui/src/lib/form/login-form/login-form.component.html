<h1>Login Form</h1>
<div class="error-message" *ngIf="error">{{ error }}</div>
<tabu-dynamic-form
  [formModel]="formModel"
  [data]="credentials"
  (saveEntry)="onlogin($event)"
></tabu-dynamic-form>
