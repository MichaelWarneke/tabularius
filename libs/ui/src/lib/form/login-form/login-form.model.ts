import { ICredentialsAbstract } from '@tabularius/shared/models';
import {
  FormControlTextbox,
  FormControlDate,
  Validators
} from '@tabularius/shared/dynamic-form';

export class CredentialFormModel implements ICredentialsAbstract {
  email = new FormControlTextbox('', [Validators.required]);
  userName = new FormControlTextbox('', []);
  password = new FormControlTextbox('', [Validators.required]);
}

export class FormSetup {
  model = new CredentialFormModel();
  submitButtonText = 'Login';
}
