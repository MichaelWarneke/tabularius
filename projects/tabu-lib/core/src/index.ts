/*
 * Public API Surface of tabu-lib
 */

export * from './lib/core/core.module';
export * from './lib/material/material.module';


