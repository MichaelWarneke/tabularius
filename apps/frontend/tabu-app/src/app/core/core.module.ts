import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { TabuAppSidenavModule } from '@tabularius/ui';

@NgModule({
  declarations: [],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    TabuAppSidenavModule
  ],
  exports: [TabuAppSidenavModule]
})
export class CoreModule {}
