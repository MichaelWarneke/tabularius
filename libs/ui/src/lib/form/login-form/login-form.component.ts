import { Component, OnInit, Output, EventEmitter } from '@angular/core';
import { FormSetup } from './login-form.model';
import { ICredentials } from '@tabularius/shared/models';

@Component({
  selector: 'tabu-login-form',
  templateUrl: './login-form.component.html',
  styleUrls: ['./login-form.component.scss']
})
export class LoginFormComponent implements OnInit {
  @Output() login = new EventEmitter<any>();

  formModel = new FormSetup();
  credentials: ICredentials = { email: '', password: '', userName: '' };

  constructor() {}

  ngOnInit() {}
  onlogin(credentials: ICredentials) {
    this.login.emit(credentials);
  }
}
