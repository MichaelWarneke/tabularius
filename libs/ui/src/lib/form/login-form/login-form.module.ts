import { NgModule } from '@angular/core';
import { LoginFormComponent } from './login-form.component';
import { SharedModule } from '@tabularius/shared/ui';
import { DynamicFormModule } from '@tabularius/shared/dynamic-form';

@NgModule({
  declarations: [LoginFormComponent],
  imports: [SharedModule, DynamicFormModule],
  exports: [LoginFormComponent]
})
export class LoginFormModule {}
