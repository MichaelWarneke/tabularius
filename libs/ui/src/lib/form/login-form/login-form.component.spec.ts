import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginFormComponent } from './login-form.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('LoginFormComponent', () => {
  let component: LoginFormComponent;
  let fixture: ComponentFixture<LoginFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [LoginFormComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
