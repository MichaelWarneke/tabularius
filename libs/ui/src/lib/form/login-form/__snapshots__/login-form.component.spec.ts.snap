// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormComponent should render 1`] = `
<tabu-login-form
  credentials={[Function Object]}
  error=""
  formModel={[Function FormSetup]}
  login={[Function EventEmitter]}
>
  <h1>
    Login Form
  </h1><tabu-dynamic-form />
</tabu-login-form>
`;
